### PR TITLE
feat: add `docs/guides/docker-automation.md`

### PR DESCRIPTION
**Pull request by grutrissheit**
Added a guide for learning how to use Docker to setup the common automation media stack (torrent client, usenet client, VPN with a killswitch, Sonarr/Radarr, Jackett).

### CODE REVIEW

File: docs/guides/docker.md [hunk 1]
```diff
@@ -0,0 +1,256 @@
+---
+label: Docker automation
+order: -2
+description: Learn how to deploy a Docker stack for common media automation software.
```
Comment (Ravencentric) [inline docs/guides/docker.md:4]: we don't use periods in headings/titles/etc. This is embed heading so remove it

File: docs/guides/docker.md [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+[here](https://docs.docker.com/engine/install/ubuntu/#install-using-the-repository), after which they can proceed onto the
+[post-installation steps](https://docs.docker.com/engine/install/linux-postinstall).
+
+## Setting up a Docker network
+
+Now we'll be setting up a network so that our containers can communicate with each other. This step's quite simple, it's just `docker network create guide`. A list of all your current networks can be seen with `docker network ls`.
+
+## Generating a Wireguard configuration file
+
+For maximum privacy and security we'll be tunnelling our usenet and torrent clients through a VPN. Radarr, Sonarr and Jackett however won't go
```
Comment (Ravencentric) [inline docs/guides/docker.md:29]: Probably worth mentioning Prowlarr?

File: docs/guides/docker.md [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+      - PUID=1000
+      - PGID=1000
+      - TZ=Europe/London
+      - PEERPORT=1000 # Replace this with the port that you forwarded during the `Generating a Wireguard configuration file` step.
+      - TRANSMISSION_RPC_PORT=9091
+    depends_on:
+      - wireguard
+    volumes:
+      - './transmission:/config'
+      - '/home/user/data/torrents:/downloads'
```
Comment (Ravencentric) [inline docs/guides/docker.md:136]: this is bad, breaks hardlinking. The same parent folder (e.g `/home/user/data/:/data` must be mounted across your automation stack. You'll do the differentiating (`/usenet`, `/torrents`, etc) in the Web UI

File: docs/guides/docker.md (hunk 1 above)
Comment (grutrissheit) [reply docs/guides/docker.md]: removed.

File: docs/guides/docker.md (hunk 2 above)
Comment (grutrissheit) [reply docs/guides/docker.md:30]: added a mention of prowlarr as a jackett alternative.

File: docs/guides/docker.md (hunk 3 above)
Comment (grutrissheit) [reply docs/guides/docker.md]: fixed and made the volume mounts consistent across all containers.

File: docs/guides/docker.md [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+
+## Deploying everything using `docker compose`
+
+Now that we've gotten everything ready we can finally set up the stack. Paste the following into a file called `docker-compose.yml`. We'd recommend
+this file being a directory that is purely for this Docker stack, such as `/home/user/projects/automation/`.
+
+We would also recommend sorting your media-related folders in the following way for ease of use:
+- `/home/user/data/torrents/` - where you torrent client will save files.
+- `/home/user/data/usenet` - where your usenet client will save files.
+- `/home/user/data/media` - where Sonarr and Radarr will hardlink files so that they can be read by Plex/Jellyfin/etc. later on.
```
Comment (Ravencentric) [inline docs/guides/docker.md:46]: seperate this into `/home/user/data/media/tv` and `/home/user/data/media/movie` for sonarr and radarr respectively

```
home/
└── user/
    └── data/
        ├── usenet
        ├── torrents
        └── media/
            ├── tv
            └── movies
```

**Pull request merged**